services/core-api/app/api/document_generation/resources/now_document.py

```diff
@@ -1,35 +1,79 @@
 import os
-import uuid
-from flask import current_app, send_file, request
-from flask_restplus import Resource, fields
-from werkzeug.exceptions import NotFound, BadRequest
+from flask import current_app, request, Response, stream_with_context
+from flask_restplus import Resource
+from werkzeug.exceptions import BadRequest, NotFound
 from app.extensions import api, cache
+from datetime import datetime
+
 from app.api.utils.resources_mixins import UserMixin
 from app.api.utils.access_decorators import requires_role_view_all, requires_role_edit_permit
 from app.api.utils.custom_reqparser import CustomReqparser
 from app.api.constants import NOW_DOCUMENT_DOWNLOAD_TOKEN
+
+from app.api.mines.documents.models.mine_document import MineDocument
+from app.api.now_applications.models.now_application_identity import NOWApplicationIdentity
 from app.api.now_applications.models.now_application_document_type import NOWApplicationDocumentType
+from app.api.now_applications.models.now_application_document_xref import NOWApplicationDocumentXref
 from app.api.services.document_generator_service import DocumentGeneratorService
+from app.api.services.document_manager_service import DocumentManagerService
 
 
 class NoticeOfWorkDocumentResource(Resource, UserMixin):
     @api.doc(
-        description='Returns the generated document associated with the received token.',
+        description=
+        'Returns the generated document associated with the received token and pushes it to the \
+        Document Manager and associates it with the Notice of Work.',
         params={'token': 'uuid4 token to execute the planned file generation for that token'})
     def get(self):
+
+        # Ensure that the token is valid
```

Comment:
We should also make a comment in the code here about our requirement to have a valid user's token to generate the document. And describe the required order of operations to satisfy saving the generated now: first hit the generate document and then immediately hit download resource.

Reply:
Added comment